experiments/scan_classes/sequences.py

```diff
@@ -0,0 +1,261 @@
+                    next_pulse['combine_index'] = next_pulse['combine_total'] - 1
+                    next_pulse['isarepeat'] = False
+                    next_pulse['intra_pulse_start_time'] = next_pulse['pulse_timing_us'] \
+                        - pulse['pulse_timing_us']
+                    next_pulse['pulse_timing_us'] = pulse['pulse_timing_us']
+                    next_pulse['combined_pulse_index'] = pulse['combined_pulse_index']
+                    next_pulse_index = next_pulse_index + 1
+
+                    if next_pulse_index == len(self.pulses):  # last pulse has been combined - end
+                        combine_pulses = False
+                        total_combined_pulses = pulse['combined_pulse_index'] + 1
+                else:
+                    combine_pulses = False
+                    if 'iscombined' not in pulse.keys():
+                        pulse['iscombined'] = False
+
+            this_pulse_index = next_pulse_index
+            combined_pulse_index += 1
+            # Jump ahead depending how many pulses we've combined.
+            # Combined pulse list is a list of lists of pulses, 
+            #   combined as to how they are sent as samples to 
+            #   driver.
+
+        self.total_combined_pulses = total_combined_pulses
+
+        # max combine_total is = power_divider for this sequence.
+        power_divider = 1
+        for pulse in self.pulses:
+            if 'combine_total' in pulse:
+                if pulse['combine_total'] > power_divider:
+                    power_divider = pulse['combine_total']
+        self.power_divider = power_divider  # TODO: is this best option. Should powers be adjusted
+        # by frequency, other factors outside of the specific sequence.
+
+        # Find repeats
+        for pulse_index in range(1, len(self.pulses)):
+            pulse = self.pulses[pulse_index]
+            if pulse['iscombined']:
+                if pulse['combine_index'] != 0:
+                    # this pulse will be a repeat if the pulse with combine_index of 0 is a repeat.
+                    first_pulse_in_combination_index = pulse_index - pulse['combine_index']
+                    pulse['isarepeat'] = self.pulses[first_pulse_in_combination_index]['isarepeat']
+                elif not self.pulses[pulse_index -1]['iscombined']:
+                    # pulse['combine_index'] = 0, lastpulse iscombined = False.
+                    pulse['isarepeat'] = False  # the last pulse must be combined in some way as well.
+                elif pulse['combine_total'] != self.pulses[pulse_index -1]['combine_total']:
+                    pulse['isarepeat'] = False  # must have same number of slices combined.
+                else:
+                    last_pulse_index = pulse_index -1
+                    last_pulse = self.pulses[last_pulse_index]
+                    # get subset for only this combined pulse
+                    combined_pulse_1 = []
+                    combined_pulse_2 = []
```

Comment:
I'm not sure what you're saying here, is it to change how I combine pulses altogether or just how I find repeats? Are you saying to make a combined_pulses list (superceding the original pulses list)? If so that would have to be a list of lists of dictionaries then. combined_pulses =
 [combined_pulse_1, combined_pulse_2, ...] where combined_pulse_1 = [pulse1, pulse2, ...] and pulse1={'pulse_timing_us' : ... , 'combine_total' : ... , ...} ?? Except for I don't think I would need to initialize new dictionaries in that case. In either case you can't determine isarepeat value until after combined.